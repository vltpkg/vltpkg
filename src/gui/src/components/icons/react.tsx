export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
