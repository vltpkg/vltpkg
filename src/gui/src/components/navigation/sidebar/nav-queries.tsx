import { useEffect, useState } from 'react'
import {
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarMenuBadge,
} from '@/components/ui/sidebar.jsx'
import { Star } from 'lucide-react'
import { useGraphStore } from '@/state/index.js'

const SidebarQueryNav = () => {
  const activeRoute = useGraphStore(state => state.activeRoute)
  const savedQueries = useGraphStore(state => state.savedQueries)
  const [queryCount, setQueryCount] = useState<number>(0)

  useEffect(() => {
    if (savedQueries) {
      setQueryCount(savedQueries.length)
    } else {
      setQueryCount(0)
    }
  }, [savedQueries])

  return (
    <SidebarMenu>
      <SidebarMenuItem>
        <SidebarMenuButton
          isActive={activeRoute === '/queries'}
          asChild
          tooltip="Saved queries"
          className="data-[active=true]:bg-neutral-800 data-[active=true]:text-white data-[active=true]:dark:bg-neutral-100 data-[active=true]:dark:text-black">
          <a
            href="/queries"
            className={`text-foreground ${activeRoute === '/queries' ? '' : ''}`}>
            <Star />
            <span>Queries</span>
            <SidebarMenuBadge
              className={
                activeRoute === '/queries' ?
                  'text-white dark:text-black'
                : ''
              }>
              {queryCount !== 0 ? queryCount : undefined}
            </SidebarMenuBadge>
          </a>
        </SidebarMenuButton>
      </SidebarMenuItem>
    </SidebarMenu>
  )
}

export { SidebarQueryNav }
