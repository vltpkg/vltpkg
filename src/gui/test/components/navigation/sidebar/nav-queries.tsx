import { describe, it, expect, afterEach, vi } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { SidebarQueryNav } from '@/components/navigation/sidebar/nav-queries.jsx'
import { useGraphStore as useStore } from '@/state/index.js'

vi.mock('@/components/ui/sidebar.jsx', () => ({
  SidebarMenu: 'gui-sidebar-menu',
  SidebarMenuButton: 'gui-sidebar-menu-button',
  SidebarMenuItem: 'gui-sidebar-menu-item',
  SidebarMenuBadge: 'gui-sidebar-menu-badge',
}))

vi.mock('lucide-react', () => ({
  Star: 'gui-star-icon',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

describe('AppSidebar query nav', () => {
  it('renders correctly', () => {
    const Container = () => {
      return <SidebarQueryNav />
    }

    const { container } = render(<Container />)
    expect(container.innerHTML).toMatchSnapshot()
  })
})
