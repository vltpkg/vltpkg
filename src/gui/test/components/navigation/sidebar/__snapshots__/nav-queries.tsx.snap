// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppSidebar query nav > renders correctly 1`] = `

<gui-sidebar-menu>
  <gui-sidebar-menu-item>
    <gui-sidebar-menu-button
      isactive="false"
      aschild="true"
      tooltip="Saved queries"
      classname="data-[active=true]:bg-neutral-800 data-[active=true]:text-white data-[active=true]:dark:bg-neutral-100 data-[active=true]:dark:text-black"
    >
      <a
        href="/queries"
        class="text-foreground "
      >
        <gui-star-icon>
        </gui-star-icon>
        <span>
          Queries
        </span>
        <gui-sidebar-menu-badge classname>
        </gui-sidebar-menu-badge>
      </a>
    </gui-sidebar-menu-button>
  </gui-sidebar-menu-item>
</gui-sidebar-menu>

`;
